BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -137,7 +141,11 @@ uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
 void CAN1_RX0_IRQHandler(void)
 {
     CAN_Receive(CAN1, CAN_FIFO0, &RxMessage);
-
+    for(int i = 0; i < 8; i++){
+        CANFIFO[CANFIFOHEAD][i] = RxMessage.Data[i]; //store data in FIFO
+        (++CANFIFOHEAD)%CANFIFOMAX; //increment FIFO Head
+    }
```

Comment:
Make sure you check that the FIFO is not full, or at least clearly specify what will happen if it is. Right now this would drop the least recent messge in favor of an incoming message. Is this what we want? Maybe @ClarkPoon could weigh in on this.

Reply:
I did this using an error status variable and comparing the locations of the head and tail. I think it will work but there might be something wrong with the logic.